rts/Lua/LuaMatrixImpl.h

```diff
@@ -0,0 +1,164 @@
+	void Scale(const float x, const float y, const float z);
+	void Scale(const float s);
+
+	void RotateRad(const float rad, const float x, const float y, const float z);
+	void RotateDeg(const float deg, const float x, const float y, const float z) { RotateRad(deg * math::DEG_TO_RAD, x, y, z); };
+
+	void RotateRadX(const float rad) { RotateRad(rad, 1.0f, 0.0f, 0.0f); };
+	void RotateRadY(const float rad) { RotateRad(rad, 0.0f, 1.0f, 0.0f); };
+	void RotateRadZ(const float rad) { RotateRad(rad, 0.0f, 0.0f, 1.0f); };
+
+	void RotateDegX(const float deg) { RotateRad(deg * math::DEG_TO_RAD, 1.0f, 0.0f, 0.0f); };
+	void RotateDegY(const float deg) { RotateRad(deg * math::DEG_TO_RAD, 0.0f, 1.0f, 0.0f); };
+	void RotateDegZ(const float deg) { RotateRad(deg * math::DEG_TO_RAD, 0.0f, 0.0f, 1.0f); };
+
+	// NB: sol::this_state is passed implicitly by sol. It's nothing more than a POD type carrying lua_State*, thus no need to pass it by reference
+	bool UnitMatrix(const unsigned int unitID, const sol::optional<bool> mult, sol::this_state L);
+	bool UnitPieceMatrix(const unsigned int unitID, const unsigned int pieceNum, const sol::optional<bool> mult, sol::this_state L);
+
+	bool FeatureMatrix(const unsigned int featureID, const sol::optional<bool> mult, sol::this_state L);
+	bool FeaturePieceMatrix(const unsigned int featureID, const unsigned int pieceNum, const sol::optional<bool> mult, sol::this_state L);
+
+	bool ProjectileMatrix(const unsigned int projID, const sol::optional<bool> mult, sol::this_state L);
+
+	void ScreenViewMatrix(const sol::optional<bool> mult);
+	void ScreenProjMatrix(const sol::optional<bool> mult);
+	void ScreenViewProjMatrix(const sol::optional<bool> mult);
+
+	void SunViewMatrix(const sol::optional<bool> mult);
+	void SunProjMatrix(const sol::optional<bool> mult);
+	void SunViewProjMatrix(const sol::optional<bool> mult);
+
+	void Ortho(const float left, const float right, const float bottom, const float top, const float near, const float far, const sol::optional<bool> mult);
+	void Frustum(const float left, const float right, const float bottom, const float top, const float near, const float far, const sol::optional<bool> mult);
+
+	void CameraViewMatrix(const sol::optional<unsigned int> cameraIdOpt, const sol::optional<bool> mult);
+	void CameraProjMatrix(const sol::optional<unsigned int> cameraIdOpt, const sol::optional<bool> mult);
+	void CameraViewProjMatrix(const sol::optional<unsigned int> cameraIdOpt, const sol::optional<bool> mult);
+	void CameraViewInverseMatrix(const sol::optional<unsigned int> cameraIdOpt, const sol::optional<bool> mult);
+	void CameraProjInverseMatrix(const sol::optional<unsigned int> cameraIdOpt, const sol::optional<bool> mult);
+	void CameraViewProjInverseMatrix(const sol::optional<unsigned int> cameraIdOpt, const sol::optional<bool> mult);
+	void CameraBillboardMatrix(const sol::optional<unsigned int> cameraIdOpt, const sol::optional<bool> mult);
+
+	tuple16f GetAsScalar()
```

Comment:
btw, this and some other methods can also have a [[nodiscard]] attribute as it makes no sense to invoke them without using the return value

Reply:
These methods are effectively only used from Lua.
I'm not sure if `[[nodiscard]]` makes any sense here, because it enforces nothing on the Lua side.